### PR TITLE
[KAS-8]: make Sign Up Screen

### PR DESCRIPTION
**Pull request by RiffatiKhan**
## Jira Story Link:
[KAS-8](https://leezeit.atlassian.net/browse/KAS-8)

**Description**:
1) Sign up screen with one logo, one label, four text fields and one button.
2) First and last name text fields only accept letters.
3) Next button enable only when user enter a valid email.
3) Alert shows message if any text field is empty.
4) Password text field is limited to six letters and its secure entry.
5) Key board dismiss when user tap anywhere on the screen.
6) Screen is scrollable so keyboard doesn’t  cover the text fields and button.
7) Screen is designed using a storyboard.

**How to Test this**:
Click the button in main storyboard.

**ScreenShot & Gif** 

Screen Shot 2022-09-06 at 7 22 33 PM

### CODE REVIEW

File: LeezeIt/LeezeIt/TextField.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import Foundation
+import UIKit
+
+// adding max lenght attribute on interface builder
+private var __maxLengths = [UITextField: Int]()
+
+extension UITextField {
+    @IBInspectable var maxLength: Int {
+        get {
+            guard let l = __maxLengths[self] else {
```
Comment (haroonmaqsoodch) [inline LeezeIt/LeezeIt/TextField.swift]: your story says "Restrict the user to enter more than 6 letters in password." change the logic of maxLength

Comment (haroonmaqsoodch) [inline LeezeIt/LeezeIt/TextField.swift]: what is L? fix your naming convention please

File: LeezeIt/LeezeIt/TextField.swift [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+               return 150 // (global default-limit. or just, Int.max)
+            }
+            return l
+        }
+        set {
+            __maxLengths[self] = newValue
```
Comment (haroonmaqsoodch) [inline LeezeIt/LeezeIt/TextField.swift]: why are you use underscores before maxLength?

File: LeezeIt/LeezeIt/ViewController.swift [hunk 3]
```diff
@@ -9,11 +9,10 @@ import UIKit
 
 class ViewController: UIViewController {
 
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        // Do any additional setup after loading the view.
+    @IBAction func gotoNextScreen() {
+        
+        let storyboard = UIStoryboard.init(name: "CreateAccount", bundle: nil)
+        let nextVc = storyboard.instantiateViewController(withIdentifier: "CreateAccountViewController") as! CreateAccountViewController
```
Comment (haroonmaqsoodch) [inline LeezeIt/LeezeIt/ViewController.swift]: try not to force downcast

File: LeezeIt/LeezeIt/CreateAccountViewController.swift [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        return true
+    }
+    
+    func alertForMissingText(messageAlert: String) {
+        let alert = UIAlertController(title: "Something missing!", message: messageAlert, preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: "Ok", style: .cancel))
+        self.present(alert, animated: true,completion: nil)
+    }
+    
+    func showAlertOfMissingTextInTextField() {
```
Comment (haroonmaqsoodch) [inline LeezeIt/LeezeIt/CreateAccountViewController.swift:72]: I was not able to see alert, could attach a screen record when alert appears?

File: LeezeIt/LeezeIt/TextField.swift (hunk 1 above)
Comment (RiffatiKhan) [reply LeezeIt/LeezeIt/TextField.swift]: Resolved

Comment (RiffatiKhan) [reply LeezeIt/LeezeIt/TextField.swift]: Resolved

File: LeezeIt/LeezeIt/TextField.swift (hunk 2 above)
Comment (RiffatiKhan) [reply LeezeIt/LeezeIt/TextField.swift]: Resolved

File: LeezeIt/LeezeIt/ViewController.swift (hunk 3 above)
Comment (RiffatiKhan) [reply LeezeIt/LeezeIt/ViewController.swift]: Resolved

File: LeezeIt/LeezeIt/CreateAccountViewController.swift (hunk 4 above)
Comment (RiffatiKhan) [reply LeezeIt/LeezeIt/CreateAccountViewController.swift:72]: https://user-images.githubusercontent.com/111387186/188982110-39eb22d9-e551-4148-9f41-1611666285ec.mp4

**Review by haroonmaqsoodch — COMMENTED** [review]
After typing the email and password button stays disabled untill clicked twice, check that also

**Review by Harrisali93 — APPROVED** [review]

**Review by Harrisali93 — APPROVED** [review]

**Review by Harrisali93 — APPROVED** [review]

**Review by Harrisali93 — APPROVED** [review]

**Review by arfhan786 — APPROVED** [review]

**Pull request merged**